File: question_two/flightschedule.py

class Flight:
    def __init__(self, id, priority, type):
        self.id = id
        self.priority = priority
        self.type = type  
    def __lt__(self, other):
        return self.priority < other.priority
def schedule_flights(flights):
    heapq.heapify(flights)
    while flights:
        flight = heapq.heappop(flights)
        print(f"Processing flight {flight.id} ({flight.type}) with priority {flight.priority}")
flights = [
    Flight("VIP001", 1, "landing"),
    Flight("EMG002", 0, "landing"),
    Flight("REG003", 3, "takeoff"),
    Flight("REG004", 2, "landing")
]
schedule_flights(flights)

File: question_two/flightschedule.js
class Flight {
    constructor(id, priority, type) {
        this.id = id;
        this.priority = priority;
        this.type = type;
    }
}
function scheduleFlights(flights) {
    flights.sort((a, b) => a.priority - b.priority);
    flights.forEach(flight => {
        console.log(`Processing flight ${flight.id} (${flight.type}) with priority ${flight.priority}`);
    });
}
const flights = [
    new Flight("VIP001", 1, "landing"),
    new Flight("EMG002", 0, "landing"),
    new Flight("REG003", 3, "takeoff"),
    new Flight("REG004", 2, "landing")
];
scheduleFlights(flights);